From c5d1a01aa9fc718a8b77ead0ea06afc8ef3ff3e6 Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Thu, 14 Mar 2013 15:20:37 +0800
Subject: [PATCH 05/10] ENGR00254281: add support to build differnt target
 images

Introduce a simple way to build ubifs for nand device base on Android
build system.
For those boards support booting from both NAND and eMMC/SD, this patch
provide a way to build out nand images with just add a build option to
enable build ubifs or ext4.
e.g make BUILD_TARGET_FS=ubifs will build image for NAND device
      make BUILD_TARGET_FS=ext4 will build image for eMMC/SD
	  if don't add BUILD_TARGET_FS when build, the build system will
	  build image according default config in BoardConfig.mk file.

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 core/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index 027669f..2ec1662 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -658,11 +658,16 @@ endif
 recovery_resources_private := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery/res))
 recovery_resource_deps := $(shell find $(recovery_resources_common) \
   $(recovery_resources_private) -type f)
+
+ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
+recovery_fstab := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery_nand.fstab))
+else
 ifdef TARGET_RECOVERY_FSTAB
 recovery_fstab := $(TARGET_RECOVERY_FSTAB)
 else
 recovery_fstab := $(strip $(wildcard $(TARGET_DEVICE_DIR)/recovery.fstab))
 endif
+endif
 # Named '.dat' so we don't attempt to use imgdiff for patching it.
 RECOVERY_RESOURCE_ZIP := $(TARGET_OUT)/etc/recovery-resource.dat
 
-- 
1.8.0

